Documentation/security/network/encryption-wireguard.rst

```diff
@@ -295,6 +295,75 @@ options:
++----------------+-------------------+----------------------+-----------------+
+| Node           | remote Pod        | any                  | node-to-node    |
++----------------+-------------------+----------------------+-----------------+
+| Node           | remote Node       | any                  | node-to-node    |
++----------------+-------------------+----------------------+-----------------+
+| **Services**                                                                |
++----------------+-------------------+----------------------+-----------------+
+| Pod            | remote Pod via    | any                  | default         |
+|                | ClusterIP Service |                      |                 |
++----------------+-------------------+----------------------+-----------------+
+| Pod            | remote Pod via    | Socket LB            | default         |
+|                | non-ClusterIP     |                      |                 |
+|                | Service (e.g.,    |                      |                 |
+|                | NodePort)         |                      |                 |
++----------------+-------------------+----------------------+-----------------+
+| Pod            | remote Pod via    | kube-proxy           | node-to-node    |
+|                | non-ClusterIP     |                      |                 |
+|                | Service           |                      |                 |
++----------------+-------------------+----------------------+-----------------+
+| Client outside | Pod via Service   | KPR,                 | default         |
+| cluster        |                   | tunneling,           |                 |
+|                |                   | without DSR          |                 |
++----------------+-------------------+----------------------+-----------------+
+| Client outside | Pod via Service   | native routing       | node-to-node    |
+| cluster        |                   |                      |                 |
++----------------+-------------------+----------------------+-----------------+
+| Pod            | remote Pod via L7 | L7 Proxy / Ingress   | default         |
+|                | Proxy or L7       |                      |                 |
+|                | Ingress Service   |                      |                 |
++----------------+-------------------+----------------------+-----------------+
+| **Egress Gateway**                                                          |
++----------------+-------------------+----------------------+-----------------+
+| Pod            |Egress Gateway node| Egress Gateway       | default         |
++----------------+-------------------+----------------------+-----------------+
+| Egress Gateway | Pod               | Egress Gateway       | default         |
+| node           |                   | without XDP          |                 |
++----------------+-------------------+----------------------+-----------------+
+
```

Comment:
So the assumption is that if something isn't in the above table, then it is not going to be encrypted, which I think makes things easy to reason about.

However, should we still explicitly call out in a paragraph somewhere the few very common cases we intentionally don't encrypt? For example `local-pod to local-pod` or `local-pod to local-node`? Or do you think that would just make it more confusing again?

Reply:
> However, should we still explicitly call out in a paragraph somewhere the few very common cases we intentionally don't encrypt? For example local-pod to local-pod or local-pod to local-node? Or do you think that would just make it more confusing again?

I wanted to keep the table as concise as possible. Also, personally, `* -> local *` doesn't make any sense. Anyway, no strong opinion on it. If you think it's worth, then I can add it. WDYT?

Reply:
Yes, I don't think we should add it to the table per se, rather I would have imagined a sentence or so in a "limitations" section or something. I also share your opinion that it doesn't make sense to encrypt such traffic.

In any case, I think this needs to be part of this PR. Marking as resolved.